src/Features/LanguageServer/Protocol/RequestDispatcher.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer
+{
+    /// <summary>
+    /// Aggregates handlers for the specified language and dispatches LSP requests
+    /// to the appropriate handler for the request.
+    /// </summary>
+    internal class RequestDispatcher
```

Comment:
this was basically renamed and modified to use the providers from http://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/AbstractRequestHandlerProvider.cs,39070e776b7d80f3